Foundation/NSURLSession/NSURLSession.swift

```diff
@@ -234,6 +239,11 @@ open class URLSession : NSObject {
         let c = URLSession._Configuration(URLSessionConfiguration: configuration)
         self._configuration = c
         self.multiHandle = _MultiHandle(configuration: c, workQueue: workQueue)
+        // registering all the protocol classes with URLProtocol
+        let _ = URLProtocol.registerClass(_HTTPURLProtocol.self)
```

Comment:
I am wondering if we want to register the native protocol classes every time a new URLSession is created. This can be a one-time activity.